app/src/main/java/com/github/se/signify/ui/screens/challenge/CreateAChallengeScreen.kt

```diff
@@ -51,17 +41,24 @@ fun CreateAChallengeScreen(
       topBar = { BackButton { navigationActions.goBack() } },
       content = { padding ->
         Column(
-            modifier = Modifier.fillMaxSize().padding(padding).padding(16.dp),
+            modifier =
+                Modifier.fillMaxSize()
+                    .padding(padding)
+                    .padding(16.dp)
+                    .testTag("CreateAChallengeContent"),
             horizontalAlignment = Alignment.CenterHorizontally,
             verticalArrangement = Arrangement.Top) {
               // Title
-              Text(text = "Select a Friend to Challenge", fontSize = 24.sp)
+              Text(
+                  text = "Select a Friend to Challenge",
+                  fontSize = 24.sp,
+                  modifier = Modifier.testTag("ChallengeTitle"))
 
               Spacer(modifier = Modifier.height(16.dp))
 
               // List of Friends
```

Comment:
The friend list integration looks good. As a UX improvement, you could display a 'No friends available' message when the list is empty to guide the user.